Src/Sound/SoundStream.cpp

```diff
@@ -0,0 +1,204 @@
+// A Sound Stream is the thing that actually plays audio data like PGE::Sound through the speakers.
+#include "SoundStream.h"
+
+#include <Math/Math.h>
+#include <String/String.h>
+#include <Misc/FilePath.h>
+
+#include "PGESound.h"
+
+#include <../../Engine/Libraries/SDL2/include/SDL_audio.h>
+#include <../../Engine/Libraries/SDL2/include/SDL.h>
+
+#include <fstream>
+#include <map>
+#include <ogg/ogg.h>
+#include <vorbis/codec.h>
+//#include <vorbis/vorbisenc.h>
+#include <vorbis/vorbisfile.h>
+
+class OggStream
+{
+    int mSerial;
+    ogg_stream_state mState;
+    int mPacketCount;
+};
+typedef std::vector<int, OggStream*> StreamMap;
+
+SoundStream::SoundStream() {
+    printf("HELLO FROM NEW SOUND STREAM\n");
+}
+
+SDL_AudioDeviceID dev;
+SDL_AudioSpec baseChannel;
+
+
+ogg_int16_t convbuffer[4096]; /* take 8k out of the data segment, not the stack */
+int convsize = 4096;
+void SoundStream::playTestTone() {
+    printf("HELLO FROM TEST TONE\n");
+    PGE::String filename = "C:/Users/Pyro/source/repos/scpcb/Content/SCPCB/SFX/SCP/914/Refining.ogg";
```

Comment:
Don't use absolute paths to your local machine.

Reply:
Thinking back on it, PGE::FilePath was giving me a file relative to /x86 so I changed it, thinking it did not have the assets in it. That was actually what I should have expected. Woops lol.